daemon/algod/api/server/v2/handlers.go

```diff
@@ -74,6 +74,57 @@ type NodeInterface interface {
 	RemoveParticipationKey(account.ParticipationID) error
 }
 
+func roundToPtrOrNil(value basics.Round) *uint64 {
+	if value == 0 {
+		return nil
+	}
+	result := uint64(value)
+	return &result
+}
+
+func convertParticipationRecord(record account.ParticipationRecord) generated.ParticipationKey {
+	participationKey := generated.ParticipationKey{
+		Id:                  record.ParticipationID.String(),
+		Address:             record.Account.String(),
+		EffectiveFirstValid: nil,
+		EffectiveLastValid:  nil,
+		LastVote:            nil,
+		LastBlockProposal:   nil,
+		LastStateProof:      nil,
```

Comment:
setting all these to nil isn't really needed, as nil would be the default value.